src/Sparrow.Server/Platform/Pal.cs

```diff
@@ -16,67 +16,18 @@ public static unsafe class Pal
 
         static Pal()
         {
-            var toFilename = LIBRVNPAL;
-            string fromFilename;
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
-            {
-                if (RuntimeInformation.ProcessArchitecture != Architecture.Arm &&
-                    RuntimeInformation.ProcessArchitecture != Architecture.Arm64)
-                {
-                    fromFilename = Environment.Is64BitProcess ? $"{toFilename}.linux.x64.so" : $"{toFilename}.linux.x86.so";
-                    toFilename += ".so";
-                }
-                else
-                {
-                    fromFilename = Environment.Is64BitProcess ? $"{toFilename}.arm.64.so" : $"{toFilename}.arm.32.so";
-                    toFilename += ".so";
-                }
-            }
-            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
-            {
-                fromFilename = Environment.Is64BitProcess ? $"{toFilename}.mac.x64.dylib" : $"{toFilename}.mac.x86.dylib";
-                // in mac we are not : `toFilename += ".so";` as DllImport doesn't assume .so nor .dylib by default
-            }
-            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-            {
-                var win7 = PlatformDetails.IsWindows8OrNewer ? "" : "7";
-
-                fromFilename = Environment.Is64BitProcess ? $"{toFilename}.win{win7}.x64.dll" : $"{toFilename}.win{win7}.x86.dll";
-                toFilename += ".dll";
-            }
-            else
-            {
-                throw new NotSupportedException("Not supported platform - no Windows/OSX/Linux is detected ");
-            }
-
-            var retries = 0;
-            while (true)
-            {
-                var e = CopyPalFile(toFilename, fromFilename);
-                if (e == null)
-                    break;
-
-                if (++retries < 10)
-                    continue;
-
-                var msg = $"Cannot copy {fromFilename} to {toFilename}";
-                if (e is IOException)
-                {
-                    throw new IOException($"{msg}, make sure appropriate {toFilename} to your platform architecture exists in Raven.Server executable folder.", e);
-                }
-
-                throw new InvalidOperationException($"{msg}.", e);
-            }
-
-            PalFlags.FailCodes rc = PalFlags.FailCodes.None;
+            PalFlags.FailCodes rc;
             int errorCode;
             try
             {
-                var palver = rvn_get_pal_ver();
-                if (palver != 0 && palver != PAL_VER)
+                var mutator = PlatformDetails.IsWindows8OrNewer == false ? (Func<string,string>)ToWin7DllName : default;
```

Comment:
instead of using mutator we can use PlatformDetails inside the DynamicNativeLibraryResolver, we will not need the mutator at all then (maybe rename to SodiumNativeLibraryResolver?)

Reply:
The problem is that we don't have `win7` for Sodium (and won't for zstd)